### PR TITLE
feat: add clang

### LINKED ISSUE #255: Docker image missing clang
Hi ! Since the project hasn't been open-sourced yet, I've added all the dependencies used in the actual project to the demo's Cargo.toml. Currently, I'm stuck on the missing clang issue, and I'm not sure if other dependencies are also missing. I hope the next version will be able to build this demo：https://github.com/MagicGordon/compile_error

### PR DESCRIPTION
**Pull request by dj8yfo**
this resolves https://github.com/near/cargo-near/issues/255

tested on https://github.com/dj8yfo/compile_error/blob/3dd4741a53572a79497807b5d8806a3997f85c75/Cargo.toml#L15-L16

built from this image https://github.com/dj8yfo/cargo-near-image/blob/677b43f0778f43c939499429e3d566589ce98045/Dockerfile#L13

### CODE REVIEW

**Comment by Canvinus** [comment]
Do we also need `clang` for the `cargo-near-commit` Dockerfile?

**Review by frol — APPROVED** [review]
I was under assumption that Rustup ships clang as well since it heavily uses LLVM and it would make sense to use the matching LLVM for C dependencies, but it is not the case: https://github.com/rust-lang/rust/issues/56371

@dj8yfo @Canvinus we may also consider adding clang++ (for C++ modules), though I cannot think of contract dependencies that would require C++ (in non-contract environment, there is only one C++ dependency that I know - RocksDB)

@race-of-sloths score 3

**Comment by race-of-sloths** [comment]
@dj8yfo Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!
Great job! Usain Bolt jealous of your results!

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://race-of-sloths.com/images/bot-banner-dark.svg">
    <source media="(prefers-color-scheme: light)" srcset="https://race-of-sloths.com/images/bot-banner-light.svg">
    Shows inviting banner with latest news.
</picture>](https://race-of-sloths.com/latest-news)

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/dj8yfo?type=bot&pr=SourceScan/cargo-near-image/4&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/dj8yfo?type=bot&pr=SourceScan/cargo-near-image/4&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/dj8yfo)





<details><summary>Current status: <i>executed</i></summary>

| Reviewer | Score |
|--------|--------|
| @frol  | 3 |


Your contribution is much appreciated with a final score of 3!
You have received 35 (30 base + 15% lifetime bonus) Sloth points for this contribution

@frol received 25 Sloth Points for reviewing and scoring this pull request.


</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/dj8yfo)

For maintainers:
- Score pull requests that participate in the Race of Sloths and receive a reward
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Invite contributor** `@race-of-sloths invite` to invite the contributor to participate in a race or include it, if it's already a runner.
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

**Pull request merged**
